DataFormats/ForwardDetId/test/testHGCDetId.cc

```diff
@@ -0,0 +1,129 @@
+#include "DataFormats/ForwardDetId/interface/HGCScintillatorDetId.h"
+#include "DataFormats/ForwardDetId/interface/HGCSiliconDetId.h"
+#include "DataFormats/DetId/interface/DetId.h"
+
+#include <cmath>
+#include <iomanip>
+#include <iostream>
+#include <map>
+#include <string>
+
+void testCell(int type) {
+
+  int    N  = (type == 0) ? 12 : 8;
+  const int waferu(0), waferv(0), layer(1), zside(1);
+  std::map<std::pair<int,int>,int> triggers;
+  int    ntot(0);
+  for (int u=0; u<2*N; ++u) {
+    for (int v=0; v<2*N; ++v) {
+      if (((v-u) < N) && (u-v) <= N) {
+	HGCSiliconDetId id(DetId::HGCalEE,zside,type,layer,waferu,waferv,u,v);
+	std::cout << "ID " << std::hex << id.rawId() << std::dec << " " << id 
+		  << " Trigger: " << id.triggerCellU() << ":"
+		  << id.triggerCellV() << std::endl;
+	std::pair<int,int> trig = id.triggerCellUV();
+	std::map<std::pair<int,int>,int>::iterator itr = triggers.find(trig);
+	if (itr == triggers.end()) {
+	  triggers[trig] = 0;
+	  itr            = triggers.find(trig);
+	}
+	++(itr->second);
+	++ntot;
+      }
+    }
+  }
+  std::cout << "Total of " << ntot << " cells in type " << type << " with "
+	    << triggers.size() << " trigger cells" << std::endl;
+  int k(0);
+  for (std::map<std::pair<int,int>,int>::iterator itr = triggers.begin();
+       itr != triggers.end(); ++itr) {
+    std::cout << "Trigger[" << k << "]  (" << (itr->first).first << ":"
+	      << (itr->first).second << ")  " << itr->second << std::endl;
+    ++k;
+  }
+}
+
+void testWafer(int layer, double rin, double rout) {
+  const double waferSize(167.4408);
```

Comment:
there are a lot of hardcoded numbers in this function and the next one. I know they're supposed to be standalone test functions, but how much of this can be obtained from geometry/topology information? It might be worthwhile to make this into an `EDAnalyzer` in that case.

Reply:
These are arbitary #'s and I do not want it to be dependent on Geometry at this stage. For a more meaningful test with realistic geometry we shall add a test in the Geometry package